frontend/packages/dev-console/src/components/taskruns/logs/taskRunLogSnippet.ts

```diff
@@ -0,0 +1,58 @@
+import { Condition, PLRTaskRunStep, TaskRunKind } from '../../../utils/pipeline-augment';
+import { taskRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
+
+type ErrorDetails = {
+  title: string;
+};
+
+type ErrorDetailsWithLogName = ErrorDetails & {
+  containerName: string;
+  podName: string;
+};
+type ErrorDetailsWithStaticLog = ErrorDetails & {
+  staticMessage: string;
+};
+
+export type TaskRunErrorDetails = ErrorDetailsWithLogName | ErrorDetailsWithStaticLog;
+
+const joinConditions = (conditions: Condition[]) =>
+  conditions.map((condition) => condition.message).join('\n') || 'Unknown failure condition';
+
+export const getLogSnippet = (taskRun: TaskRunKind): TaskRunErrorDetails => {
```

Comment:
I'm confused... you refactored the PipelineRunLogSnippet to be more able to work with both, why is this still the old (almost)duplicated format?

It appears this is also used by `TaskRunDetailsErrorLog`. My point in my last review was to merge these two, which I thought you were doing with the changes to `pipelineRunLogSnippet.ts`.

What am I missing?

Reply:
For PLR -> Check the PLR status -> search for the relevant TaskRun which failed -> return correct message based on TaskRun
For TaskRun -> Check TaskRun status -> return correct message based on the TaskRun

The last step in both is the same. Hence, I have written that code in the `log-util` file and rest is separate.

Reply:
For `PipelineRunDetailsErrorLog` vs `TaskRunDetailsErrorLog`, if you go by the previous comments, it was decided that the two be kept separate. Earlier we were using the same component.